File: job_finder_app/lib/Model/Job.dart
import 'package:job_finder_app/Model/Employer.dart';
import 'package:job_finder_app/Model/Specialization.dart';

class Job {
  int? id;
  int? employerId;
  String? title;
  String? description;
  String? location;
  int? salary;
  int? specializationId;
  String? deadline;
  Specialization? specialization;
  Employer? employer;

  Job({
    this.id,
    this.employerId,
    this.title,
    this.description,
    this.location,
    this.salary,
    this.specializationId,
    this.deadline,
    this.specialization,
    this.employer,
  });

  Job.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    employerId = json['employer_id'];
    title = json['title'];
    description = json['description'];
    location = json['location'];
    salary = json['salary'];
    specializationId = json['specialization_id'];
    deadline = json['deadline'];
    specialization =
        json['specialization'] != null
            ? new Specialization.fromJson(json['specialization'])
            : null;
    employer =
        json['employer'] != null
            ? new Employer.fromJson(json['employer'])
            : null;
  }
}

File: JobFinder/app/Models/Job.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Job extends Model
{
    use HasFactory;
    public $timestamps = false;
    protected $hidden = [
        'created_at',
        'updated_at',
    ];
    protected $fillable = [
        'employer_id',
        'title',
        'description',
        'location',
        'salary',
        'specialization_id',
        'deadline',
    ];

    public function employer()
    {
        return $this->belongsTo(Employer::class, 'employer_id');
    }

    public function specialization()
    {
        return $this->belongsTo(Specialization::class);
    }

    public function applications()
    {
        return $this->hasMany(JobApplication::class);
    }
}
